SquirrelInputController.m

```diff
@@ -151,6 +158,11 @@ - (BOOL)handleEvent:(NSEvent*)event client:(id)sender
           modifiers & OSX_CAPITAL_MASK);
         if (rime_keycode) {
           int rime_modifiers = osx_modifiers_to_rime_modifiers(modifiers);
+          // revert non-modifier function keys' FunctionKeyMask (FwdDel, Navigations, F1..F19)
+          if (keyCode >= 0x60 || keyCode == 0x50 || keyCode == 0x4f ||
```

Comment:
`keyCode >= 0x60` is not  a closed range.

Reply:
Reverting the function key mask will result in another issue: the `Hyper` modifier cannot compose with all function keys. For example, `Hyper+F1` is a valid combination but not possible to trigger using Fn key.

Reply:
> `keyCode >= 0x60` is not a closed range.

Depending on the keyboard, all remaining functional keys (e.g. F25) are in this range, without a definite upper bound.

Reply:
> Reverting the function key mask will result in another issue: the `Hyper` modifier cannot compose with all function keys. For example, `Hyper+F1` is a valid combination but not possible to trigger using Fn key.

`Hyper+F1` might be valid on linux or other OS, but `Fn+F1` is not valid on MacOS, since it triggers exactly the same event as `F1`, both have the `Fn` modifier. But there are still plenty other key combinations available, including `Fn` itself. Specifically, `Fn+｀` is ideal to avoid keybinding conflicts.

Reply:
> Reverting the function key mask will result in another issue: the `Hyper` modifier cannot compose with all function keys. For example, `Hyper+F1` is a valid combination but not possible to trigger using Fn key.

On the other hand, it's fairly uncommon to set keybindings involving function keys. `Ctrl+F1` is very rare, so it wouldn't make much sense to set `Hyper+F1` to begin with. The only ones that may be troublesome are `Fn+Up/Down/Left/Right/return/delete` as these key events do not exist but are instead `Fn+PageUp/PageDown/Home/End/enter/forwardDelete`